### PR TITLE
SC-194 Channel Groups Sorting

### PR DESCRIPTION
**Pull request by KulinduC**
Relies on OpenXDA changes: https://github.com/GridProtectionAlliance/openXDA/pull/336 and Gemstone changes: https://github.com/GridProtectionAlliance/gpa-gemstone/pull/306

### CODE REVIEW

**Comment by KulinduC** [comment]
image
image
image

File: Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/ChannelGroup/ByChannelGroup.tsx
Comment (gcsantos-gpa) [inline Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/ChannelGroup/ByChannelGroup.tsx:1]: Searchbar should be dispatching ChannelGroupViewSlice not the ChannelGroupSlice

File: Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/Store/Store.ts [hunk 1]
```diff
@@ -192,6 +194,7 @@ const store = configureStore({
         MATLABAnalyticAssetType: MATLABAnalyticAssetTypeSlice.Reducer,
         TrendChannels: TrendChannelSlice.Reducer,
         ChannelTemplate: ChannelTemplateSlice.Reducer,
+        ChannelGroupView: ChannelGroupViewSlice.Reducer
```
Comment (gcsantos-gpa) [inline Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/Store/Store.ts:197]:
```suggestion
        ChannelGroupView: ChannelGroupViewSlice.Reducer,
```

File: Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/ChannelGroup/ByChannelGroup.tsx [hunk 2]
```diff
@@ -133,29 +133,28 @@ const ChannelGroups: Application.Types.iByComponent = (props) => {
...
                         Key={'Description'}
                         AllowSort={true}
                         Field={'Description'}
                         HeaderStyle={{ width: 'auto' }}
                         RowStyle={{ width: 'auto' }}
                     > Description
                     </ReactTable.Column>
-                    <ReactTable.Column<SystemCenter.Types.ChannelGroup>
+                    <ReactTable.Column<LocalSC.ChannelGroupView>
                         Key={'Items'}
```
Comment (gcsantos-gpa) [inline Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/ChannelGroup/ByChannelGroup.tsx:153]:
```suggestion
                        Key={'ItemCount'}
```
Arrow for sort indication is based on key not field, so key must also be ItemCount

**Review by gcsantos-gpa — CHANGES_REQUESTED** [review]

**Review by gcsantos-gpa — APPROVED** [review]

**Pull request merged**